content/articles/automatic-sms-verification-with-sms-retriever-api/index.md

```diff
@@ -0,0 +1,339 @@
+
+        //uncomment this to generate your app hash string. You can view the hash string on your log cat when you run the app
+        /* val appSignatureHelper = SignatureHelper(this)
+        Log.d("SIGNATURE",appSignatureHelper.appSignature.toString())*/
+
+        initSmsListener()
+    }
+    
+     override fun onStart() {
+        super.onStart()
+        EventBus.getDefault().register(this)
+    }
+
+    override fun onStop() {
+        EventBus.getDefault().unregister(this)
+        super.onStop()
+    }
+    
+    private fun initSmsListener() {
+        smsClient.startSmsRetriever()
+            .addOnSuccessListener {
+                Toast.makeText(
+                    this, "Waiting for sms message",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }.addOnFailureListener { failure ->
+                Toast.makeText(
+                    this, failure.localizedMessage,
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+    }
+
+    @Subscribe
+    fun onReceiveSms(retrievalEvent: RetrievalEvent) {
+        val code: String =
+            StringUtils.substringAfterLast(retrievalEvent.message, "is").replace(":", "")
+                .trim().substring(0, 4)
+
+        runOnUiThread {
+            if (!retrievalEvent.timedOut) {
+                binding.editText.setText(code)
+            } else {
+                Toast.makeText(this, "Failed", Toast.LENGTH_SHORT).show()
+            }
+        }
+        initSmsListener()
+    }
+}
+```
+
+### Computing your app's hash string
+To generate the hash string, you can use the following methods:
+- Use [Play App Signing](https://support.google.com/googleplay/android-developer/answer/9842756?visit_id=637672247631770776-2285078183&rd=1)
+- Use `SignatureHelper class`. This class will help to generate our app's hash string. After using this class to obtain the hash string, always remove it.
+
+```kotlin
+class SignatureHelper(context: Context?) :
+    ContextWrapper(context) {
+    val appSignature: ArrayList<String>
+        get() {
+            val appCodes = ArrayList<String>()
+            try {
+                val myPackageName = packageName
+                val myPackageManager = packageManager
+                val signatures =
+                    myPackageManager.getPackageInfo(
+                        myPackageName,
+                        PackageManager.GET_SIGNATURES
+                    ).signatures
+
+                for (signature in signatures) {
+                    val hash =
+                        hash(myPackageName, signature.toCharsString())
+                    if (hash != null) {
+                        appCodes.add(String.format("%s", hash))
+                    }
+                }
+            } catch (e: PackageManager.NameNotFoundException) {
+                Log.d(
+                    TAG,
+                    "Package not found",
+                    e
+                )
+            }
```

Comment:
Remember to work on the arrangement of your code!

Reply:
@Ericgacoki Already done the changes requested

Reply:
Looks good :rocket: